Fix/docker setup

cardosofede opened the pull request:
This PR fixes the errors when trying to run the project from Docker.
Main changes:

1. Remove general Dockerfile and add specific Dockerfile for Frontend and Backend.
2. Make the URL of the Backend configurable via env variables.
3. Refactor docker-compose.yml file to run redis + backend + frontend.
4. Minor fix in .env placeholders since the values were requested when starting the product --> is necessary to add error handling in the app.

donatienthorez commented:
This fix is really good and helped me getting it running with Docker. Is there any blocker from merging it? 🤔

albertoecf commented:
Same here! It ran successfully on my instance. @nfcampos / @eyurtsev  is there any blocker for merging it ?

cardosofede commented:
Thanks @donatienthorez @albertoecf for the feedback!

nfcampos commented:
The main blocker for merging this is the removal of the root docker file, that can be used for deployments so it shouldn't be remove imo

cardosofede commented:
Hello @nfcampos, thank you for your feedback.

I've adjusted the Docker setup for better modularity. The main Dockerfile, previously used for the backend, has been moved to the backend directory. A new Dockerfile is introduced for the frontend. This separation allows independent development and deployment of each component.

For deployment, you can use `docker-compose.yml`, which builds both the backend and frontend from their respective Dockerfiles.

The steps to get started are:
1. Clone the repository.
2. Run `docker compose up`.
3. Wait for the Docker images for both frontend and backend to build and Redisearch to be downloaded.
4. The system is ready for use.

I've dropped you a message on Slack to discuss this approach in more detail. I'm open to feedback and willing to make any adjustments necessary to facilitate the merging of this PR.

Thanks, Fede.

nfcampos commented on frontend/package.json:
```diff
@@ -4,7 +4,7 @@
   "type": "module",
   "scripts": {
-    "dev": "vite",
+    "dev": "vite --host",
```
what does this do?

cardosofede replied:
The --host option allows the server to be accessed over the network, not just localhost. it's necessary when running via docker.
When running without --host and exposing the port from the container to the local machine we weren't able to see the output. With that argument, the Frontend was exposed.

fengtality commented:
I installed OpenGPTs from both source and this Docker setup. 

Docker is much simpler since `docker compose up` works and you don't need to deal with Redis, installing dependencies, and adding placeholder values for env variables.
